### PR TITLE
Feat date picker

### PR DESCRIPTION
**Pull request by twumm**
# Description
Add date picker
Update event card with author name

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Change status
- [x] Complete, tested, ready to review and merge
- [] Complete, but not tested (may need new tests)
- [] Incomplete/work-in-progress, PR is for discussion/feedback

# How Has This Been Tested?
Tested the entire app to make sure events are returned

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation, if relevant
- [x] I have run the app using my feature and ensured that no functionality is broken
- [x] My changes generate no new warnings
- [x] There are no merge conflicts

### CODE REVIEW

File: components/atoms/Interests/index.js [hunk 1]
```diff
@@ -7,7 +7,7 @@ export default function InterestSelect({ interest, setInterest }) {
       <select
         value={interest}
         onChange={(e) => { setInterest(e.target.value); }}
-        placeHolder="Select Interest"
+        placeholder="Select Interest"
```
Comment (josenriagu) [inline components/atoms/Interests/index.js:10]: seems like we have implemented same fix here. haha

File: components/layout/App.js [hunk 2]
```diff
@@ -30,14 +35,15 @@ const App = ({ events }) => {
...
               {
-              events.length > 0
-                ? events.map(el => <EventCard key={el.scrapedEventId} el={el} />)
-                : <Loader />
+                // console.log(events)
+                (!fetchEventsError && fetchEventsRequesting)
+                  ? <Loader />
+                  : (!fetchEventsRequesting && typeof events === 'object' && events.length > 0)
+                    ? events.map(el => <EventCard key={el.scrapedEventId} el={el} />)
+                    : <Heading color="red">There are no events currently. Please try the search feature.</Heading>
```
Comment (josenriagu) [inline components/layout/App.js]: Awezzzzome job! on adding the conditions

File: components/layout/App.js [hunk 3]
```diff
@@ -30,14 +35,15 @@ const App = ({ events }) => {
...
               }
             </div>
-            <div id="calendar">
-              <Calendar fullscreen={false} />
-            </div>
+            <Calendar />
```
Comment (josenriagu) [inline components/layout/App.js:44]: fullscreen prop not required? @twumm

**Comment by twumm** [comment]
[Date picker](https://trello.com/c/rLSSfZgJ/56-date-picker)

File: components/layout/App.js [hunk 4]
```diff
@@ -48,11 +54,15 @@ const App = ({ events }) => {
 
 App.propTypes = {
   // eslint-disable-next-line react/forbid-prop-types
```
Comment (josenriagu) [inline components/layout/App.js]: please remove this line if you have already disabled this eslint warning for the entire file

File: components/molecules/Searchbar/index.js [hunk 5]
```diff
@@ -19,6 +19,10 @@ const SearchBar = ({ doFetchEvent }) => {
   const handleClick = () => {
     const sendCountry = location.country.toLowerCase().replace(' ', '-');
     const sendCity = location.city === 'Abuja Federal Capital Territory' ? 'abuja' : location.city.toLowerCase().replace(' ', '-');
+    localStorage.setItem('userSearchCountry', sendCountry);
+    localStorage.setItem('userSearchCity', sendCity);
+    localStorage.setItem('userEventType', interest);
+    console.log(sendCity, sendCountry, interest)
```
Comment (josenriagu) [inline components/molecules/Searchbar/index.js]: console may not be needed anymore and can be safely removed??

File: components/molecules/Searchbar/index.js [hunk 6]
```diff
@@ -19,6 +19,10 @@ const SearchBar = ({ doFetchEvent }) => {
   const handleClick = () => {
     const sendCountry = location.country.toLowerCase().replace(' ', '-');
     const sendCity = location.city === 'Abuja Federal Capital Territory' ? 'abuja' : location.city.toLowerCase().replace(' ', '-');
+    localStorage.setItem('userSearchCountry', sendCountry);
+    localStorage.setItem('userSearchCity', sendCity);
+    localStorage.setItem('userEventType', interest);
```
Comment (josenriagu) [inline components/molecules/Searchbar/index.js:24]: I guess these localStorage variables are going to be reused?

File: components/organisms/Calendar/index.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+  const onClick = (event) => {
+    const { userSearchCity, userSearchCountry, userInitialCountry, userInitialCity } = localStorage;
+    console.log(userSearchCity, userSearchCountry, userInitialCountry, userInitialCity)
+    const userCity = userSearchCity || userInitialCity;
+    const userCountry = userSearchCountry || userInitialCountry;
+    const startDate = moment(event._d).format('YYYY-MM-DD');
+    const endDate = moment(event._d).add(1, 'd').format('YYYY-MM-DD');
+    const eventType = 'all';
+    // doFetchCalendarEvent(userCountry, userCity, eventType, startDate, endDate);
```
Comment (josenriagu) [inline components/organisms/Calendar/index.js]: doFetchCalendarEvent not implemented anymore?

File: redux/actions/events.js [hunk 8]
```diff
@@ -37,3 +37,27 @@ export const doFetchEvent = (userCountry, userCity, eventType) => dispatch => {
...
+      endDate,
+    }),
+  };
+  // utilize fetch here since axios won't work here. refer to: https://stackoverflow.com/questions/56721660/how-to-fix-error-request-failed-with-status-code-404-in-axios-next-js
+  fetch('https://comcalstaging.herokuapp.com/api/v1/event/fetch-date', config)
+    .then(res => res.json())
+    .then(data => {
+      dispatch(fetchEventsSuccess(data));
+    })
+    .catch(error => dispatch(fetchEventsError(error)));
```
Comment (josenriagu) [inline redux/actions/events.js:62]: great job!

**Review by josenriagu — COMMENTED** [review]
Good job buddy!
Could you please check the review comments and reply/fix accordingly

File: components/atoms/Interests/index.js (hunk 1 above)
Comment (twumm) [reply components/atoms/Interests/index.js:10]: lol. No wahala

File: components/layout/App.js (hunk 4 above)
Comment (twumm) [reply components/layout/App.js]: Noted and done!

File: components/molecules/Searchbar/index.js (hunk 6 above)
Comment (twumm) [reply components/molecules/Searchbar/index.js:24]: Yes. Check the Calendar organism component

File: components/organisms/Calendar/index.js (hunk 7 above)
Comment (twumm) [reply components/organisms/Calendar/index.js]: Implemented now.

File: components/molecules/Searchbar/index.js (hunk 5 above)
Comment (twumm) [reply components/molecules/Searchbar/index.js]: Yes. An apparent oversight. Thanks!

**Pull request merged**